pullrequests/symmetric_tree/step1.go

```diff
@@ -0,0 +1,35 @@
+//lint:file-ignore U1000 Ignore all unused code
+package symmetrictree
+
+type TreeNode struct {
+	Val   int
+	Left  *TreeNode
+	Right *TreeNode
+}
+
+/*
+レビュワーの方へ：
+	- このコードは既にGoの標準のフォーマッタで整形済みです。演算子の周りにスペースがあったりなかったりしますが、これはGoのフォーマッタによるもので、優先順位の高い演算子の周りにはスペースが入らず、低い演算子の周りには入るようになっています。https://qiita.com/tchssk/items/77030b4271cd192d0347
+*/
+
+/*
+初見で解くことができなかったので、他の解法を見て解いた。再帰の練習が足りてなさそう。
+*/
+func isSymmetricRecursive(root *TreeNode) bool {
+	// 問題的にはrootはnilにならないのでこの処理は必要ないが
+	// 一応nilのときはtrue扱いにすると決めて書いた
+	if root == nil {
+		return true
+	}
+	return isMirror(root.Left, root.Right)
+}
+
+func isMirror(left, right *TreeNode) bool {
+	if left == nil && right == nil {
+		return true
+	}
+	if left == nil || right == nil {
+		return false
+	}
+	return left.Val == right.Val && isMirror(left.Left, right.Right) && isMirror(left.Right, right.Left)
```

Comment:
少し1行が長くなっていて、かつ「現在着目しているノードの話」と「子ノードの話」が含まれた条件になっているので、個人的には分割したほうがスッキリする気がします。

```suggestion
        if left.Val != right.Val {
                return false
        }
	return isMirror(left.Left, right.Right) && isMirror(left.Right, right.Left)
```